pkg/prober/prober.go

```diff
@@ -0,0 +1,155 @@
+package prober
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/common/route"
+)
+
+const (
+	healthyEndpointPath  = "/-/healthy"
+	readyEndpointPath    = "/-/ready"
+	okProbeText          = "thanos %v is %v"
+	errorProbeText       = "thanos %v is not %v. Reason: %v"
+	probeErrorHTTPStatus = 500
```

Comment:
This is totally debatable but imho bad healthchecks should be [503 Service Unavailable,](https://developer.mozilla.org/fr/docs/Web/HTTP/Status/503)

Reply:
Definitely! Thanks for pointing it out